client/react/lib/index.tsx

```diff
@@ -29,75 +29,126 @@ interface setComponent {
+const FallbackBuilder = <T extends {}>({ fallback, status, ...props }: FallbackBuilderProps<T>) => {
   if (!fallback || isElement(fallback)) {
     return fallback;
   }
   const FallbackComponent = fallback as React.ComponentType<any>;
   return <FallbackComponent dynamicoStatus={status} {...props} />;
-}
+};
 
 export const dynamico = function<T = any>(
   name: string,
   { fallback = null, devMode = false, ...options }: ComponentOptions<T> = {}
 ): FunctionComponent<T> {
   return (props: T) => {
     const [Component, setComponent]: [Component, setComponent] = useState({});
-    const [status, setStatus]: [Status, setStatus] = useState<Status>({currentStatus: ComponentStatus.Loading});
-    
-    const dynamicoClient = useContext<DynamicoClient | undefined>(DynamicoContext);
-    let release = () => {};
-
-    const getComponent = async () => {
-      if (!dynamicoClient) {
-        throw `Couldn't find dynamico client in the context, make sure you use DynamicoContext.Provider`;
-      }
-      if (devMode) {
-        const devClient = new DynamicoDevClient({
-          dependencies: dynamicoClient.dependencies,
-          ...(typeof devMode === 'object' ? devMode : {}),
-          callback: (view: any) => setComponent({ view })
-        });
-
-        release = await devClient.get(name, options);
-
-        return;
-      }
-      
-      setComponent({ view: await dynamicoClient.get(name, options) });
-    };
+    const [status, setStatus]: [Status, setStatus] = useState<Status>({ currentStatus: ComponentStatus.Loading });
+
+    const { client: dynamicoClient, devClient } = useContext(DynamicoContext);
 
     useEffect(() => {
-      
-        getComponent().catch((error) => {
-          setStatus({
-            currentStatus: ComponentStatus.Error,
-            error
+      let release = () => {};
+
+      const setError = (error: Error) => setStatus({ currentStatus: ComponentStatus.Error, error });
+
+      const getComponent = async () => {
+        if (!dynamicoClient) {
+          throw `Couldn't find dynamico client in the context, make sure you use DynamicoContext.Provider`;
+        }
+
+        if (devMode || devClient) {
+          const devOptions = typeof devMode === 'object' ? devMode : {};
+
+          const client =
+            devClient ||
+            new DynamicoDevClient({
+              dependencies: dynamicoClient.dependencies,
+              ...devOptions
+            });
+
+          let usingFallbackComponent = false;
+
+          release = client.get(name, {
+            interval: devOptions.interval,
+            ...options,
+            callback: async (err, view: any) => {
+              if (!err) {
+                usingFallbackComponent = false;
+                return setComponent({ view });
+              }
+
+              console.warn(`failed getting component ${name} from dev server`, err);
+
+              if (!devMode) {
```

Comment:
How can I get to this code path with `!devMode`?

Reply:
So the approach here is that if I have `devMode === false` I'm in global dev mode... mode... and so I should treat all components in dev mode? I would prefer to be more explicit... how about returning from the provider some property that says `devModeOverride` or `globalDevMode`?

Reply:
if you're in global dev mode, you get a dev client from the provider.
if the component is explicitly using dev mode we won't try getting the component from remote server and render the fallback
if the component is not explicitly using dev mode we try getting the component from remote server, and if that fails too render fallback

Reply:
also added - if you explicitly set `devMode: false` on a component, it will not use global devMode